### PR TITLE
Update OpenTelemetry plugin to use latest packages

### LINKED ISSUE #1610: opentelemetry cannot be provided with a globally registered TracerProvider
Using `@envelop/opentelemetry`, there's an option to pass a custom provider to it, but it's required that it conforms to `BasicTracerProvider` and not `TracerProvider`. Furthermore I noticed that `@opentelemetry/tracing` is used which has been deprecated in favor of `@opentelemetry/sdk-trace-base`. 

Ideally `@envelop/opentelemetry` would acquire the globally registered tracer through `@opentelemetry/api`.

```ts
import * as opentelemetry from '@opentelemetry/api';

const tracer = opentelemetry.trace.getTracer(serviceName);
```

Alternatively to be a non-breaking change, `trace` should be capable of providing `TracerProvider`.

```ts
import * as opentelemetry from '@opentelemetry/api';

const getEnveloped = envelop({
  plugins: [
    useEngine({ parse, validate, specifiedRules, execute, subscribe }),
    // ... other plugins ...
    useOpenTelemetry(
      {
        resolvers: true, // Tracks resolvers calls, and tracks resolvers thrown errors
        variables: true, // Includes the operation variables values as part of the metadata collected
        result: true // Includes execution result object as part of the metadata collected
      },
      opentelemetry.trace.getTracerProvider()
    )
  ]
})
```

The simplest solution seems to widen the API to accept any `TracerProvider` and to additionally upgrade to use `@opentelemetry/sdk-trace-base`.

### PR DESCRIPTION
**Pull request by darren-west**
## Description

Update to the latest open telemetry packages switching to using `@opentelemetry/sdk-trace-base` 

Closes https://github.com/n1ru4l/envelop/issues/1610

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Tested manually 

## Checklist:

- [x] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

Carrying on from https://github.com/n1ru4l/envelop/pull/1611 to get this merged.

### CODE REVIEW

File: .changeset/loud-swans-serve.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@envelop/opentelemetry": minor
```
Comment (shift) [inline .changeset/loud-swans-serve.md]: Linting is failing on this file. Another change set which passed looks like:
```
---
'@envelop/validation-cache': patch
---

fix: validation cache for ESM environments
```

So my guess is its failing on the double quotes and is expecting single quotes. Probably best to also update the message to be conventional.

Comment (darren-west) [reply .changeset/loud-swans-serve.md]: Hi @shift thanks I have updated this now.

**Review by shift — APPROVED** [review]

File: packages/plugins/opentelemetry/package-lock.json
Comment (n1ru4l) [inline packages/plugins/opentelemetry/package-lock.json:1]: Can you please delete this file? We are using `yarn` workspaces, so you should ran `yarn install` in the root of the repository instead.

Comment (darren-west) [reply packages/plugins/opentelemetry/package-lock.json:1]: Oops I didnt mean to push that. Removed thanks.

Comment (darren-west) [reply packages/plugins/opentelemetry/package-lock.json:1]: Oops I didnt mean to push that. Removed thanks.

**Pull request merged**